apps/laboratory/tests/shared/pages/ModalPage.ts

```diff
@@ -226,4 +228,51 @@ export class ModalPage {
   async closeModal() {
     await this.page.getByTestId('w3m-header-close')?.click?.()
   }
+
+  async updateEmail(mailsacApiKey: string) {
+    const email = new Email(mailsacApiKey)
+    const newEmailAddress = email.getNewEmailAddress(this.emailAddress)
+
+    await this.page.getByTestId('account-button').click()
+    await this.page.getByTestId('w3m-account-email-update').click()
+    await this.page.getByTestId('wui-email-input').locator('input').focus()
+    await this.page.getByTestId('wui-email-input').locator('input').fill(newEmailAddress)
+
+    // Clear messages before putting new email
+    await email.deleteAllMessages(this.emailAddress)
+    await this.page.getByTestId('wui-email-input').locator('input').press('Enter')
+
+    // Wait until the next screen appears
+    await expect(this.page.getByText('Enter the code we sent')).toBeVisible({
+      timeout: 10_000
+    })
+    const confirmCurrentEmail = await this.page.getByText(this.emailAddress).isVisible()
```

Comment:
nitpick: would be nice to replace this with getById

Reply:
in this case, i need to verify if the screen contains the current email or the new email. Getting the component by ID will always return true